common/lib/xmodule/xmodule/tests/test_video.py

```diff
@@ -695,30 +695,39 @@ class VideoCdnTest(unittest.TestCase):
     """
     Tests for Video CDN.
     """
-    @patch('requests.get')
-    def test_get_video_success(self, cdn_response):
+    def test_rewrite_video_url_success(self):
         """
         Test successful CDN request.
         """
-        original_video_url = "http://www.original_video.com/original_video.mp4"
-        cdn_response_video_url = "http://www.cdn_video.com/cdn_video.mp4"
-        cdn_response_content = '{{"sources":["{cdn_url}"]}}'.format(cdn_url=cdn_response_video_url)
-        cdn_response.return_value = Mock(status_code=200, content=cdn_response_content)
-        fake_cdn_url = 'http://fake_cdn.com/'
+        original_video_url = "http://www.originalvideo.com/original_video.mp4"
+        fake_cdn_url = 'http://fakecdn.com/'
+        cdn_response_video_url = fake_cdn_url + "original_video.mp4"
+
         self.assertEqual(
-            get_video_from_cdn(fake_cdn_url, original_video_url),
+            rewrite_video_url(fake_cdn_url, original_video_url),
             cdn_response_video_url
         )
 
-    @patch('requests.get')
-    def test_get_no_video_exists(self, cdn_response):
+    def test_rewrite_video_url_invalid_url(self):
         """
         Test if no alternative video in CDN exists.
         """
-        original_video_url = "http://www.original_video.com/original_video.mp4"
-        cdn_response.return_value = Mock(status_code=404)
-        fake_cdn_url = 'http://fake_cdn.com/'
-        self.assertIsNone(get_video_from_cdn(fake_cdn_url, original_video_url))
+        original_video_url = "http://www.originalvideo.com/original_video.mp4"
+        fake_cdn_url = 'http://http://fakecdn.com/'
+        self.assertIsNone(rewrite_video_url(fake_cdn_url, original_video_url))
+
+    def test_none_args(self):
+        """
+        Ensure None args return None
+        """
+        self.assertIsNone(rewrite_video_url(None, None))
+
+    def test_emptystring_args(self):
```

Comment:
You mean combining these tests?  My preference would  be to have a single tests that tested all input for this small set, e.g., None/None,None/"",""/"",""/None
Not sure if you are cool with that or whether there is prior art.